### DISCUSSION TITLE
forge compile vs forge build

### DISCUSSION THREAD

**Discussion by devmousa** [Cyfrin/foundry-full-course-cu#657] [General]
I followed the course and write my first project in Foundry and everything went well
but I've just astounded, what is the difference between these commands ?
```bash
forge compile
forge build
```

I know that `forge compile` compiles my Solidity code into EVM bytecode, but when I tried `forge build`, it did the same, compiled my Solidity files!

does anybody know what is the difference ?
thank you in advance  ❤️

**Comment by theirrationalone** [comment]
Build: Checks the code Verifies it & gives guarantee to be error(only syntactic) free, afterwards The `Compiler|Interpreter` hands over the `bytecode` to machine as well as hands over `ABI`, `META_INFORMATION`, & `OPCODE`, to users(us) in `json` format, outputs a folder called `out`.

Compile: An Apposition for `Build` only or vice versa!

**Reply by devmousa** [reply]
I didn't get it well 😥

what did you mean with this:
Compile: An Apposition for Build only or vice versa!

**Reply by theirrationalone** [reply]
Let's say a person has two names, alice & ammy, and often people call her ammy but sometimes few people call her alice.  So alice is an apposition for ammy and same in opposite case... Ammy <-> alice.

**Reply by devmousa** [reply]
you made it clearer
thank you a lot, I get it now 😊
